### ISSUE TITLE
[Bug]: Menu button pop up options not accessible via voiceOver, and background is clickable

### ISSUE THREAD

**Issue by nikitasingla-MSFT** [microsoft/fluentui#30488]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
DEVICE USED: IOS, iphone12pro
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codesandbox.io/p/sandbox/mystifying-vaughan-ksyjjt

## Actual Behavior
When user try to activate the dropdown pop-up menu items using Voiceover 'double tap to activate', here the control is not getting activated. whereas on double tap it closes the menu pop up
The a11y box is bigger than the focus box, hence the selection is not happening.

AND THE OUTER CONTAINER IS CLICKABLE SO THATS WHY ITS NOT GETTING THE VOICE OVER CLICK

NOTE: TRIED IT ON FLUENT V9 DROPDOWN THROUGH SANDBOX LINK

## Expected Behavior
All interactive controls should be accessible and actionable using Voiceover swipe gestures and Keyboard.

SIMILAR BUG RAISED ON OUR TEAM BY A11Y TEAM: [Bug 3459615](https://domoreexp.visualstudio.com/MSTeams/_workitems/edit/3459615): [TFL- School Connection]: The 'Help' pop up menu items are not actionable using Voiceover.
  

https://github.com/microsoft/fluentui/assets/118871240/9fa03a18-d234-4c49-8bb1-8b97b2e03f1b

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by miroslavstastny** [comment]
@nikitasingla-MSFT could you please check that the codesandbox url is valid and point to a public codesandbox. It does not open for me.

**Comment by nikitasingla-MSFT** [comment]
hey @bsunderhus , i have updated the link
